python/idsse_common/test/test_grid_proj.py

```diff
@@ -0,0 +1,146 @@
+
+
+def test_from_proj_grid_spec_with_offset():
+    proj_with_offset = GridProj.from_proj_grid_spec(PROJ_SPEC_WITH_OFFSET, EXAMPLE_GRID_SPEC)
+    assert proj_with_offset._x_offset == approx(-3272536.1979)
+    assert proj_with_offset._y_offset == approx(-260290.8369)
+
+
+# transformation methods testing
+def test_map_proj_to_pixel(grid_proj: GridProj):
+    for index, proj in enumerate(EXAMPLE_PROJECTIONS):
+        pixel_x, pixel_y = grid_proj.map_proj_to_pixel(*proj)
+        assert (round(pixel_x, 6), round(pixel_y, 6)) == EXAMPLE_PIXELS[index]
+
+
+def test_map_proj_to_geo(grid_proj: GridProj):
+    for index, proj in enumerate(EXAMPLE_PROJECTIONS):
+        geo_x, geo_y = grid_proj.map_proj_to_geo(*proj)
+        assert (geo_x, geo_y) == approx_tuple(EXAMPLE_GEOS[index])
+
+
+def test_map_pixel_to_geo(grid_proj: GridProj):
+    for index, pixel in enumerate(EXAMPLE_PIXELS):
+        geo_x, geo_y = grid_proj.map_pixel_to_geo(*pixel)
+        assert (geo_x, geo_y) == approx_tuple(EXAMPLE_GEOS[index])
+
+
+def test_map_pixel_to_proj(grid_proj: GridProj):
+    for index, pixel in enumerate(EXAMPLE_PIXELS):
+        proj_x, proj_y = grid_proj.map_pixel_to_proj(*pixel)
+        assert (proj_x, proj_y) == approx_tuple(EXAMPLE_PROJECTIONS[index])
+
+
+def test_map_geo_to_proj(grid_proj: GridProj):
+    for index, geo in enumerate(EXAMPLE_GEOS):
+        proj_x, proj_y = grid_proj.map_geo_to_proj(*geo)
+        assert (proj_x, proj_y) == approx_tuple(EXAMPLE_PROJECTIONS[index])
+
+
+def test_geo_to_pixel_no_rounding(grid_proj: GridProj):
+    for index, geo in enumerate(EXAMPLE_GEOS):
+        pixel_x, pixel_y = grid_proj.map_geo_to_pixel(*geo)
+        # round result, which will not be precisely the integer that was passed
+        assert (round(pixel_x, 6), round(pixel_y, 6)) == EXAMPLE_PIXELS[index]
+
+
+def test_geo_to_pixel_floor(grid_proj: GridProj):
+    for index, geo in enumerate(EXAMPLE_GEOS):
+        pixel_x, pixel_y = grid_proj.map_geo_to_pixel(*geo, PixelRounding.FLOOR)
+        assert (pixel_x, pixel_y) == EXAMPLE_PIXELS[index]
+
+
+def test_geo_to_pixel_round(grid_proj: GridProj):
+    for index, geo in enumerate(EXAMPLE_GEOS):
+        pixel_x, pixel_y = grid_proj.map_geo_to_pixel(*geo, PixelRounding.ROUND)
+        assert (pixel_x, pixel_y) == EXAMPLE_PIXELS[index]
+
+
+def test_compound_tranformations_stay_consistent(grid_proj: GridProj):
+    # start with pixel, convert to projection
+    initial_pixel = (EXAMPLE_PIXELS[2][0], EXAMPLE_PIXELS[2][1])
+    proj_x, proj_y = grid_proj.map_pixel_to_proj(*initial_pixel)
+    initial_proj = (proj_x, proj_y)
+
+    # convert projection to geographic coordinates
+    geo_x, geo_y = grid_proj.map_proj_to_geo(proj_x, proj_y)
+    initial_geo = geo_x, geo_y
+
+    # convert geographic coordinates back to pixel, full circle, and data should be unchanged
+    pixel_x, pixel_y = grid_proj.map_geo_to_pixel(geo_x, geo_y)
+    assert (round(pixel_x, 6), round(pixel_y, 6)) == initial_pixel
```

Comment:
This will have to change when you update the round logic in grid_proj.

Reply:
Should the test now be using round_half_away, or is round correct?

Reply:
This was still using the incorrect round function, good catch. Fixed now.